src/test/java/jenkins/bouncycastle/EncodignDecodingTest.java

```diff
@@ -224,5 +226,15 @@ public void testCreationFromObjectPrivateKeyPEM() throws Exception {
 
         assertEquals(FileUtils.readFileToString(PRIVATE_KEY_PEM), FileUtils.readFileToString(pemFileNew));
     }
+    
+    
+    @Test
+    @Issue(value="JENKINS-35661") 
+    public void testReadKeyPairFromPCKS8PEM() throws Exception {
+        PEMEncodable pemEnc = PEMEncodable.read(getResourceFile("private-key-pcks8.pem"));
+        assertNotNull(pemEnc.toKeyPair());
+        assertNotNull(pemEnc.toPrivateKey());
+        assertNotNull(pemEnc.toPublicKey());
```

Comment:
@stephenc, done. To compare the reencoding I had to use the PCKS1 version of the same key, as bouncy castle always encodes these keys in PCKS1 and not in the original form we are testing PCKS8 "-----PRIVATE KEY----" instead of "-----PRIVATE RSA KEY-----"

I don't know if I understand your concern about the public exponent, I'm getting it from the private key: https://github.com/jenkinsci/bouncycastle-api-plugin/pull/8/files#diff-bd4cfea89bd3485d8ce0e1af5c973366R183